source/api_cc/src/DeepPot.cc

```diff
@@ -217,6 +230,20 @@ init (const std::string & model, const int & gpu_rank, const std::string & file_
     graph::SetDefaultDevice(str, &graph_def);
   }
   #endif // GOOGLE_CUDA
+
+  #if TENSORFLOW_USE_ROCM
+  hipGetDeviceCount(&gpu_num); // check current device environment
+  if (gpu_num > 0) {
+    options.config.set_allow_soft_placement(true);
+    options.config.mutable_gpu_options()->set_per_process_gpu_memory_fraction(0.9);
+    options.config.mutable_gpu_options()->set_allow_growth(true);
+    hipErrcheck(hipSetDevice(gpu_rank % gpu_num));
+    std::string str = "/gpu:";
+    str += std::to_string(gpu_rank % gpu_num);
+    graph::SetDefaultDevice(str, &graph_def);
+  }
```

Comment:
Couldn't this part of code be merged with the cuda GPU config setting?

Reply:
this part of code, i use "hipErrcheck" function:240

Reply:
Shouldn't we unify hipErrCheck and cudaErrCheck?

Reply:
yeah, maybe we should do this, but i don't know how to unify hipErrCheck and cudaErrCheck, they are two macros in /source/lib/include/gpu_cuda.h and /source/lib/include/gpu_rocm.h. Or your meaning is introducing the condition only when the function name appears

Reply:
>  Or your meaning is introducing the condition only when the function name appears

Yes, because the TF config for both cuda and rocm are the same (they also share the same device name `"gpu"` in TF), so we do not want to maintain two separate pieces of codes.

Reply:
> they are two macros in /source/lib/include/gpu_cuda.h and /source/lib/include/gpu_rocm.h

Are the definitions in the files the same as those in DeepPot.cc?

Reply:
how about three kinds of ErrCheck: CudaErrCheck  hipErrCheck  DeviceErrCheck, thus somewhere we need to merge the code with cuda, we use DeviceErrCheck

Reply:
OK, since this PR is already very large, we leave it as is in this PR. We should consider using a unified interface for the error checks in the next PR.